### ISSUE TITLE
SSC error

### ISSUE THREAD

**Issue by kpomazi** [TutteInstitute/fast_hdbscan#39]
The original HDBSCAN works fine for me with the library.
However, if I try to run the SSC example from this PR: https://github.com/TutteInstitute/fast_hdbscan/pull/21
I get the following error:

`from sklearn import datasets
from sklearn.preprocessing import StandardScaler
import random
import numpy as np

import fast_hdbscan

iris = datasets.load_iris()

# Function used to assign a label to only a fraction of the labels.
def sample_labels(labels, frac=0.05, seed=1):
    random.seed(seed)
    k = int(len(labels)*frac)
    indices = random.sample(list(range(0, len(labels))), k)
    partial_labels = np.array([-1]*len(labels))
    partial_labels[indices] = labels[indices]
    return(partial_labels)
    
X = iris.data
y = iris.target
X = StandardScaler().fit_transform(X)

partial_labels = sample_labels(y, frac=0.05).astype(np.int64)

# Build model
model = fast_hdbscan.HDBSCAN(semi_supervised=True).fit(X, partial_labels)
print(f"cluster_labels for Iris: {model.labels_}")`

The one thing I've changed from the example code is that I am converting the labels to np.int64 because of an other numba error.

**Comment by lmcinnes** [comment]
It seems like a corner case that has made numba unhappy. I'l try to see if I can get some time to look into this later this week.
